cosmwasm/packages/wasmi-runtime/src/exports.rs

```diff
@@ -200,6 +205,10 @@ pub unsafe extern "C" fn ecall_handle(
     sig_info: *const u8,
     sig_info_len: usize,
 ) -> HandleResult {
+    if let Err(err) = recursion_depth::increment() {
+        error!("recursion limit exceeded, can not perform query!");
```

Comment:
Would probably be more clear to change the name of `query` to `handle`

Also, can we make these `warn` instead of `error`? This is expected behaviour, so not really worth an error

Reply:
I agree with you @Cashmaney

Reply:
actually... This could even be `debug` since node runners probably won't care about this at all.
On the other hand this isn't exactly hidden because the host can theoretically track how many recursions are done per tx

Reply:
This isn't expected behavior, `init` and `handle` are always on `depth=0`.
See [my other comment](https://github.com/enigmampc/SecretNetwork/pull/517#discussion_r481971747) about adding a code comment here as well.